src/main/java/com/linecorp/armeria/server/docs/ThriftXmlDocString.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.docs;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Deque;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import javax.xml.stream.events.XMLEvent;
+
+import com.google.common.collect.Sets;
+import com.google.common.reflect.ClassPath;
+import com.google.common.reflect.ClassPath.ClassInfo;
+
+import io.netty.util.internal.ObjectUtil;
+
+/**
+ * {@link ThriftXmlDocString} is an DocString extractor for Thrift IDL XML.
```

Comment:
Can you document the conventions this class uses? From what I can tell, it searches for classes, e.g ThriftConstants.class, and strips Constants and appends .xml. If this is true, what about files that don't define any constants?

What about we define a classpath location where XML files should be packaged into,

```
resources/com/linecorp/armeria/server/thriftxml/io/mypackage/MyMessage.xml
```
and we just find all XML files inside c.l.a.s.thriftxml?